gpt4all-chat/src/chatllm.cpp

```diff
@@ -869,7 +869,7 @@ auto ChatLLM::promptInternalChat(const QStringList &enabledCollections, const LL
             auto &[promptIndex, queryStr] = *query;
             const int retrievalSize = MySettings::globalInstance()->localDocsRetrievalSize();
             emit requestRetrieveFromDB(enabledCollections, queryStr, retrievalSize, &databaseResults); // blocks
-            m_chatModel->updateSources(promptIndex, databaseResults);
+            m_chatModel->updateSources(promptIndex + startOffset, databaseResults); // start offset is only for local server messages
```

Comment:
The comment above should be read as:
> select only (local server messages) from the *current* chat session.

(As opposed to selecting local server messages from previous sessions.)

But the way this similar comment is worded implies that the comment above was misread.